test/strings/basic.jl

```diff
@@ -1093,3 +1093,29 @@ end
     @test sprint(summary, SubString("foα", 2)) == "3-codeunit SubString{String}"
     @test sprint(summary, "") == "empty String"
 end
+
+struct StringWrapper <: AbstractString
```

Comment:
There's already a `GenericString` type defined in `Test` that might be more appropriate for this?